addons/test_mail/tests/test_mail_mail.py

```diff
@@ -132,6 +133,21 @@ def test_mail_mail_send_server(self):
         self.assert_email_sent_smtp(message_from='user_2@test_2.com', emails_count=5, from_filter=self.server_domain_2.from_filter)
         self.assert_email_sent_smtp(message_from='user_1@test_2.com', emails_count=5, from_filter=self.server_domain.from_filter)
 
+    def test_mail_mail_schedule(self):
+        """Test that a mail scheduled in the past/future are sent or not"""
+        scheduled_dates = [False, '',
+                           (datetime.now() - timedelta(days=1)).strftime(DEFAULT_SERVER_DATETIME_FORMAT),
```

Comment:
I would also add a datetime that is not in server format but parsed by parser and reset back to the right format, to know it works when possible.